example/map_pt_space.cpp

```diff
@@ -0,0 +1,276 @@
+          // Temperature
+          const Real T = gTs.x(j);
+
+          // volume fractions
+          const Real lvfrac = lvfracs.x(i);
+          const Real vfrac = std::pow(10., lvfrac);
+          const Real alpha1 = vfrac;
+          const Real alpha2 = 1. - vfrac;
+
+          // microphysical densities
+          Real rho1 = singularity::robust::ratio(rhobar1, alpha1);
+          Real rho2 = singularity::robust::ratio(rhobar2, alpha2);
+
+          const Real u1 = rhobar1 * eos1.InternalEnergyFromDensityTemperature(rho1, T);
+          const Real u2 = rhobar2 * eos2.InternalEnergyFromDensityTemperature(rho2, T);
+          residuals(j, i, 0) = u1 + u2 - utot, utot;
+
+          const Real P1 = eos1.PressureFromDensityTemperature(rho1, T);
+          const Real P2 = eos2.PressureFromDensityTemperature(rho2, T);
+          residuals(j, i, 1) = P1 - P2;
+        });
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+    Kokkos::fence();
+#endif
+
+    // Now we need to copy all of this data to host so we can output it
+    std::unique_ptr<DataBox, DBDeleter> pPs_h(
+        new DataBox(Spiner::AllocationTarget::Host, nvfrac, NEOS));
+    auto Ps_h = *pPs_h;
+    portableCopyToHost(Ps_h.data(), Ps.data(), Ps.sizeBytes());
+
+    std::unique_ptr<DataBox, DBDeleter> pTrajRes_h(
+        new DataBox(Spiner::AllocationTarget::Host, nvfrac, 3));
+    auto trajRes_h = *pTrajRes_h;
+    portableCopyToHost(trajRes_h.data(), trajRes.data(), trajRes.sizeBytes());
+
+    std::unique_ptr<DataBox, DBDeleter> p_residuals_h(
+        new DataBox(Spiner::AllocationTarget::Host, nT, nvfrac, NEOS));
+    auto residuals_h = *p_residuals_h;
+    portableCopyToHost(residuals_h.data(), residuals.data(), residuals.sizeBytes());
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+    Kokkos::fence();
+#endif
+
+    FILE *file = fopen(savename.c_str(), "w");
+    PORTABLE_REQUIRE(file != nullptr, "Unable to open file");
+    // Now lets save the residuals
+    fprintf(file, "# j i T alpha_1 res_sie res_P\n");
+    for (std::size_t j = 0; j < nT; ++j) {
+      for (std::size_t i = 0; i < nvfrac; ++i) {
+        Real lvfrac = lvfracs.x(i);
+        Real vfrac = std::pow(10., lvfrac);
+        fprintf(file, "%ld %ld %.14e %.14e %.14e %.14e\n", j, i, gTs.x(j), vfrac,
+                residuals_h(j, i, 0), residuals_h(j, i, 1));
+      }
+    }
+    fclose(file);
+
+    printf("# sie and P residuals normalized by total energy\n");
+    printf("# i alpha_1 T P_1 P_2 res_alpha res_sie res_P\n");
+    for (std::size_t i = 0; i < nvfrac; ++i) {
+      Real lvfrac = lvfracs.x(i);
+      Real vfrac = std::pow(10., lvfrac);
+      printf("%ld %.14e %.14e %.14e %.14e %.14e %.14e %.14e\n", i, vfrac, Ts_h(i),
+             Ps_h(i, 0), Ps_h(i, 1), trajRes(i, 0), trajRes(i, 1), trajRes(i, 2));
+    }
+
+    eos1.Finalize();
+    eos2.Finalize();
+  }
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+  Kokkos::finalize();
+#endif
+  return 0;
+}
```

Comment:
Thanks for codifying this in an example. I wish we had done that with the EOSPAC and Spiner comparison code! We should keep it here whenever that is redone.